contrib/pyln-client/pyln/client/lightning.py

```diff
@@ -223,7 +223,7 @@ class UnixSocket(object):
     """
 
     def __init__(self, path: str):
-        self.path = path
+        self.path = str(path) if 'pathlib' in str(type(path)) else path
```

Comment:
This can simply be this:

```suggestion
        self.path = str(path)
```

This is because `str("hellworld") == "helloworld"`. On the other hand it might be sensible to keep `self.path` a `pathlib.Path` as long as possible, since it makes operating on the object much easier.